Set Cali Offsets instantly

samuelballantyne opened the pull request:
This pull request addresses an issue where calibration values do not update immediately after the trigger is pulled during the calibration process.

samuelballantyne committed:
Instant Cali Offsets

Calibration points are updated as soon as the trigger is pulled.

samuelballantyne committed:
Instant Cali Offsets

Calibration Offsets are now set as soon as you pull the trigger.

SeongGino commented on SamcoEnhanced/SamcoEnhanced.ino:
```diff
@@ -1955,60 +2000,43 @@ void ExecCalMode()
-              if(yPos > 32768/2) { yPos = yPos - 30; }
-              else { yPos = 32768/2; }
-              AbsMouse5.move(xPos, yPos);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 0;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         case Cali_Right:
-          for(xPos = 0; xPos < 32767; xPos = xPos + 30) {
-              if(xPos > 32767-31) { xPos = 32766; }
-              AbsMouse5.move(xPos, 32768/2);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32767;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         case Cali_Center:
-          for(xPos = 32767; xPos > 32768/2; xPos = xPos - 30) {
-              if(xPos < 32768/2) { xPos = 32768/2; }
-              AbsMouse5.move(xPos, 32768/2);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32768 / 2;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         default:
-          break;
+            break;
     }
 }
 
```
I know a lot of this is code I originally made, but if the only thing this function does is set init values, should we just put all these up in the ExecCalMode switch cases (after the cali values are set)? In retrospect, having this function where it is in the Calibration loop keeps tripping me up. :sweat_smile:

samuelballantyne replied:
Yeah, good point. The code has been updated to merge these switch cases into ExecCalMode.

SeongGino reviewed (commented):
Looks good, but just one little thing I had a thought about.

samuelballantyne committed:
Set Cali Offsets instantly

This pull request addresses an issue where calibration values do not update immediately after the trigger is pulled during the calibration process.

samuelballantyne committed:
Set Cali Offsets instantly

SeongGino reviewed (approved):
Looks good to me! <3
